### ISSUE TITLE
Upgrading to 12.0.3RC2 breaks header bar

### ISSUE THREAD

**Issue by Coucouf** [nextcloud/server#6542]
I just upgraded from 12.0.1 to 12.0.3RC2 and it broke the header bar.

Analysing the HTML I noticed that the header changed from an ID in <=12.0.2 :
    `<div id="header-left">`
to a class in 12.0.3RC2 :
    `<div class="header-left">`

Adding back the id="left-header" in core/templates/layout.user.php:46 fixes the issue.

**Comment by Coucouf** [comment]
Notice the barely visible icons on top of the text and home icon instead of in the header :
![image](https://user-images.githubusercontent.com/912642/30525693-b7e83d1e-9c0b-11e7-8163-3f85d1c0c56a.png)

**Comment by Coucouf** [comment]
And header-right has the same problem…
![image](https://user-images.githubusercontent.com/912642/30525704-05fe30f8-9c0c-11e7-8a69-2e6fc74c4541.png)

**Comment by MorrisJobke** [comment]
PR: https://github.com/nextcloud/server/pull/6112

@Coucouf This looks like you still have the old assets. Could you do a hard refresh in the browser (usually Shift + F5) to verify that the browser doesn't not have cached the old CSS styles. I can't reproduce this in Chrome and Firefox with the RC2.

Then the big question is, why it was not cache busted. :/

**Comment by Coucouf** [comment]
I tracked that down to the index.php/css/core/$hash-server.css using references to a former version :
    /* line 28, ${ncfolder}-12.0.1/core/css/styles.scss */
Whereas thats nc instance was pointing to a newer ${ncfolder}-12.0.3RC2 via a symlink.

I have a setup like :
    ${ncbasefolder}/nextcloud-12.0.1
    ${ncbasefolder}/nextcloud-12.0.3RC2
    ${ncbasefolder}/nextcloud -> nextcloud-12.0.3RC2
the latter being a symlink that I move on upgrade to the latest installed version folder.

It looks as if some cache mechanism didn’t notice the symlink changed and kept referencing the (still present) older nextcloud installation folder.
Moving nextcloud-12.0.1 out of ${ncbasefolder} solved the issue.

Really not straightforward though…

**Comment by Coucouf** [comment]
Worse, putting back nextcloud-12.0.1 in ${ncbasefolder} in the structure above makes the webapp again use the obsolete css… (with the symlink untouched, pointing at the latest nextcloud-12.0.3RC2)
Dark magic.

**Comment by Coucouf** [comment]
To be complete, I’ve set strong permission on all files nextcloud-$version so the www user has write access to config/ and its contents only.

**Comment by mejo-** [comment]
I can confirm that the issue still exists here (with 12.0.3 final) on a similar setup like @Coucouf described. We have `nextcloud-12.0.1` and `nextcloud-12.0.3` as separate directories next to each other and `nextcloud-current` as a symlink pointing to the latter. The Apache2 VHost has the path to `nextcloud-current` set as DocumentRoot.

Seems like #6587 is related.

**Comment by brunob** [comment]
We also got a setup based on symlinks and encounter the described bug while upgrading form 12.0.2 to 12.0.3, moving our old `nextcloud-12.0.2` to `nextcloud-12.0.2_` fixed the problem but this point out a cache bug.

**Comment by linuxrrze** [comment]
Same here: 
Current nextcloud version is linked (DocumentRoot is nextcloud-latest):
```
nextcloud-12.0.2
nextcloud-12.0.3
nextcloud-latest -> nextcloud-12.0.3
```

Strace on apache processes reports the following files to be accessed from the previous version (12.0.2) - maybe this helps to narrow down the cause:

```
nextcloud-12.0.2/apps/activity/js/activitycollection.js
nextcloud-12.0.2/apps/activity/js/activitymodel.js
nextcloud-12.0.2/apps/activity/js/activity-sidebar.json
nextcloud-12.0.2/apps/activity/js/activitytabview.js
nextcloud-12.0.2/apps/activity/js/filesplugin.js
nextcloud-12.0.2/apps/activity/js/richObjectStringParser.js
nextcloud-12.0.2/apps/comments/js/activitytabviewplugin.js
nextcloud-12.0.2/apps/comments/js/app.js
nextcloud-12.0.2/apps/comments/js/commentcollection.js
nextcloud-12.0.2/apps/comments/js/commentmodel.js
nextcloud-12.0.2/apps/comments/js/commentstabview.js
nextcloud-12.0.2/apps/comments/js/commentsummarymodel.js
nextcloud-12.0.2/apps/comments/js/filesplugin.js
nextcloud-12.0.2/apps/comments/js/merged.json
nextcloud-12.0.2/apps/files/css/detailsView.scss
nextcloud-12.0.2/apps/files/css/files.scss
nextcloud-12.0.2/apps/files/css/merged.scss
nextcloud-12.0.2/apps/files/css/mobile.scss
nextcloud-12.0.2/apps/files/css/upload.scss
nextcloud-12.0.2/apps/files/js/app.js
nextcloud-12.0.2/apps/files/js/breadcrumb.js
nextcloud-12.0.2/apps/files/js/detailfileinfoview.js
nextcloud-12.0.2/apps/files/js/detailsview.js
nextcloud-12.0.2/apps/files/js/detailtabview.js
nextcloud-12.0.2/apps/files/js/favoritesfilelist.js
nextcloud-12.0.2/apps/files/js/favoritesplugin.js
nextcloud-12.0.2/apps/files/js/fileactions.js
nextcloud-12.0.2/apps/files/js/fileactionsmenu.js
nextcloud-12.0.2/apps/files/js/fileinfomodel.js
nextcloud-12.0.2/apps/files/js/filelist.js
nextcloud-12.0.2/apps/files/js/files.js
nextcloud-12.0.2/apps/files/js/filesummary.js
nextcloud-12.0.2/apps/files/js/file-upload.js
nextcloud-12.0.2/apps/files/js/gotoplugin.js
nextcloud-12.0.2/apps/files/js/jquery.fileupload.js
nextcloud-12.0.2/apps/files/js/jquery-visibility.js
nextcloud-12.0.2/apps/files/js/keyboardshortcuts.js
nextcloud-12.0.2/apps/files/js/mainfileinfodetailview.js
nextcloud-12.0.2/apps/files/js/merged-index.json
nextcloud-12.0.2/apps/files/js/navigation.js
nextcloud-12.0.2/apps/files/js/newfilemenu.js
nextcloud-12.0.2/apps/files/js/recentfilelist.js
nextcloud-12.0.2/apps/files/js/recentplugin.js
nextcloud-12.0.2/apps/files/js/search.js
nextcloud-12.0.2/apps/files/js/sidebarpreviewmanager.js
nextcloud-12.0.2/apps/files/js/sidebarpreviewtext.js
nextcloud-12.0.2/apps/files/js/tagsplugin.js
nextcloud-12.0.2/apps/files_sharing/css/mergedAdditionalStyles.scss
nextcloud-12.0.2/apps/files_sharing/css/sharebreadcrumb.scss
nextcloud-12.0.2/apps/files_sharing/css/sharetabview.scss
nextcloud-12.0.2/apps/files_sharing/js/additionalScripts.json
nextcloud-12.0.2/apps/files_sharing/js/sharebreadcrumbview.js
nextcloud-12.0.2/apps/files_sharing/js/share.js
nextcloud-12.0.2/apps/files_sharing/js/sharetabview.js
nextcloud-12.0.2/apps/files_texteditor/css/DroidSansMono/stylesheet.scss
nextcloud-12.0.2/apps/files_texteditor/css/merged.scss
nextcloud-12.0.2/apps/files_texteditor/css/mobile.scss
nextcloud-12.0.2/apps/files_texteditor/css/style.scss
nextcloud-12.0.2/apps/files_texteditor/js/core/vendor/ace-builds/src-noconflict/ace.js
nextcloud-12.0.2/apps/files_texteditor/js/editor.js
nextcloud-12.0.2/apps/files_texteditor/js/merged.json
nextcloud-12.0.2/apps/files_texteditor/js/sidebarpreview.js
nextcloud-12.0.2/apps/files_versions/js/filesplugin.js
nextcloud-12.0.2/apps/files_versions/js/merged.json
nextcloud-12.0.2/apps/files_versions/js/versioncollection.js
nextcloud-12.0.2/apps/files_versions/js/versionmodel.js
nextcloud-12.0.2/apps/files_versions/js/versionstabview.js
nextcloud-12.0.2/apps/gallery/js/gallerybutton.js
nextcloud-12.0.2/apps/gallery/js/galleryfileaction.js
nextcloud-12.0.2/apps/gallery/js/galleryutility.js
nextcloud-12.0.2/apps/gallery/js/scripts-for-file-app.json
nextcloud-12.0.2/apps/gallery/js/slideshowcontrols.js
nextcloud-12.0.2/apps/gallery/js/slideshow.js
nextcloud-12.0.2/apps/gallery/js/slideshowzoomablepreview.js
nextcloud-12.0.2/apps/gallery/js/vendor/bigshot/bigshot-compressed.js
nextcloud-12.0.2/apps/gallery/js/vendor/dompurify/src/purify.js
nextcloud-12.0.2/apps/gallery/js/vendor/nextcloud/share.js
nextcloud-12.0.2/apps/notifications/js/app.js
nextcloud-12.0.2/apps/notifications/js/merged.json
nextcloud-12.0.2/apps/notifications/js/notification.js
nextcloud-12.0.2/apps/notifications/js/richObjectStringParser.js
nextcloud-12.0.2/apps/systemtags/js/app.js
nextcloud-12.0.2/apps/systemtags/js/filesplugin.js
nextcloud-12.0.2/apps/systemtags/js/merged.json
nextcloud-12.0.2/apps/systemtags/js/systemtagsfilelist.js
nextcloud-12.0.2/apps/systemtags/js/systemtagsinfoview.js
nextcloud-12.0.2/apps/systemtags/js/systemtagsinfoviewtoggleview.js
nextcloud-12.0.2/apps/theming/css/theming.scss
nextcloud-12.0.2/core/css/apps.scss
nextcloud-12.0.2/core/css/fixes.scss
nextcloud-12.0.2/core/css/fonts.scss
nextcloud-12.0.2/core/css/global.scss
nextcloud-12.0.2/core/css/header.scss
nextcloud-12.0.2/core/css/icons.scss
nextcloud-12.0.2/core/css/inputs.scss
nextcloud-12.0.2/core/css/mobile.scss
nextcloud-12.0.2/core/css/multiselect.scss
nextcloud-12.0.2/core/css/server.scss
nextcloud-12.0.2/core/css/share.scss
nextcloud-12.0.2/core/css/styles.scss
nextcloud-12.0.2/core/css/systemtags.scss
nextcloud-12.0.2/core/css/tooltip.scss
nextcloud-12.0.2/core/css/variables.scss
nextcloud-12.0.2/core/js/apps.js
nextcloud-12.0.2/core/js/config.js
nextcloud-12.0.2/core/js/eventsource.js
nextcloud-12.0.2/core/js/jquery.avatar.js
nextcloud-12.0.2/core/js/jquery.contactsmenu.js
nextcloud-12.0.2/core/js/jquery.ocdialog.js
nextcloud-12.0.2/core/js/js.js
nextcloud-12.0.2/core/js/l10n.js
nextcloud-12.0.2/core/js/merged-share-backend.json
nextcloud-12.0.2/core/js/merged-template-prepend.json
nextcloud-12.0.2/core/js/mimetype.js
nextcloud-12.0.2/core/js/mimetypelist.js
nextcloud-12.0.2/core/js/oc-backbone.js
nextcloud-12.0.2/core/js/oc-dialogs.js
nextcloud-12.0.2/core/js/oc-requesttoken.js
nextcloud-12.0.2/core/js/octemplate.js
nextcloud-12.0.2/core/js/placeholder.js
nextcloud-12.0.2/core/js/public/appconfig.js
nextcloud-12.0.2/core/js/select2-toggleselect.js
nextcloud-12.0.2/core/js/shareconfigmodel.js
nextcloud-12.0.2/core/js/sharedialogexpirationview.js
nextcloud-12.0.2/core/js/sharedialoglinkshareview.js
nextcloud-12.0.2/core/js/sharedialogresharerinfoview.js
nextcloud-12.0.2/core/js/sharedialogshareelistview.js
nextcloud-12.0.2/core/js/sharedialogview.js
nextcloud-12.0.2/core/js/shareitemmodel.js
nextcloud-12.0.2/core/js/share.js
nextcloud-12.0.2/core/js/sharesocialmanager.js
nextcloud-12.0.2/core/js/systemtags/merged.json
nextcloud-12.0.2/core/js/systemtags/systemtagmodel.js
nextcloud-12.0.2/core/js/systemtags/systemtagscollection.js
nextcloud-12.0.2/core/js/systemtags/systemtagsinputfield.js
nextcloud-12.0.2/core/js/systemtags/systemtags.js
nextcloud-12.0.2/core/js/systemtags/systemtagsmappingcollection.js
```

**Comment by user897435** [comment]
I can confirm this as well.

Upgraded from 12.0.2 to 12.0.3 and NO symlinks used -> header does not contain items at the right side anymore. They work (menus open), but there's no visible items in header / right. 

Some other icons look like missing as well (selecting files gives only one icon left side of Tags item) and I think the font has changed.

I pretty much ruled out cache issues as well.

**Comment by MorrisJobke** [comment]
Duplicate of #6028 
Fix for master in #7061
Fix for stable12 in #7170 (will be released with 12.0.4) - maybe test the RC: https://help.nextcloud.com/t/help-testing-nextcloud-12-0-4-rc1/23785
